sql/expression/function/substring.go

```diff
@@ -277,7 +277,7 @@ func (s *SubstringIndex) IsNullable() bool {
 }
 
 func (s *SubstringIndex) String() string {
-	return fmt.Sprintf("SUBSTRING_INDEX(%s, %s, %d)", s.str, s.delim, s.count)
+	return fmt.Sprintf("SUBSTRING_INDEX(%s, %s, %s)", s.str, s.delim, s.count)
```

Comment:
seems wrong

Reply:
s.count is of type Expression (usually a Literal)

Reply:
%d was doing something really weird with it, not sure how it ever worked